pkg/reconciler/pipeline/controller.go

```diff
@@ -599,7 +620,8 @@ func buildVertices(pl *dfv1.Pipeline) map[string]dfv1.Vertex {
 		copyVertexTemplate(pl, vCopy)
 		copyVertexLimits(pl, vCopy)
 		replicas := int32(1)
-		if pl.Status.Phase == dfv1.PipelinePhasePaused {
+		// If the desired phase is pause or we are in the middle of pausing we should not start any vertex replicas
+		if isLifecycleChange(pl) {
```

Comment:
So, I just noticed that `isLifecycleChange() == true` in the case that the oldPhase was Paused but the desiredPhase==Running. I haven't read enough of the other code yet to confirm that that's okay here ?

Reply:
That should be a `resume` pipeline case and I kept it like that for it to be handled by the 
[resume-logic](https://github.com/kohlisid/numaflow/blob/e7c32c1857a448b3399721062ff619208917ec91/pkg/reconciler/pipeline/controller.go#L796)